pkg/distributor/distributor.go

```diff
@@ -495,7 +495,8 @@ func (d *Distributor) Push(ctx context.Context, req *client.WriteRequest) (*clie
 			return nil, err
 		}
 
-		validatedSeries, err := d.validateSeries(ts, userID)
+		skipLabelNameValidation := d.cfg.SkipLabelNameValidation || req.GetSkipLabelNameValidation()
```

Comment:
@replay `SkipLabelNameValidation` was added to the config in https://github.com/cortexproject/cortex/pull/2835, based on needs of a downstream project. Do we still need it in the config or can we remote it favour of specifying it directly in the request? If it's needed in the config too, then OK, otherwise I would prefer to keep code cleaner.

Reply:
You're right, we won't need the `SkipLabelNameValidation` in the config struct anymore. 

Would it be fine for you if we remove that flag from the Distributor config in a separate PR after this one has been merged? That would help because otherwise merging this PR is going to break the down-stream CI tests of projects relying on the Distributor config flag, if we remove it in a separate PR we can first switch the down-stream users to rely on this new per-request flag and only after that we remove the Distributor config flag.

Reply:
I'm fine if you will do it in a separate PR.